ui-admin/src/study/surveys/SurveyEditorView.tsx

```diff
@@ -21,24 +24,66 @@ const SurveyEditorView = (props: SurveyEditorViewProps) => {
     onSave
   } = props
 
-  const [editedContent, setEditedContent] = useState<string>()
+  const FORM_DRAFT_KEY = getFormDraftKey({ form: currentForm })
+
   const [isEditorValid, setIsEditorValid] = useState(true)
   const [saving, setSaving] = useState(false)
-  const isSaveEnabled = !!editedContent && isEditorValid && !saving
+  const [savingDraft, setSavingDraft] = useState(false)
+
+  //Let the user know if we loaded a draft from local storage when the component first renders. It's important
+  //for them to know if the version of the survey they're seeing are from a draft or are actually published.
+  const [showLoadedDraftModal, setShowLoadedDraftModal] = useState(!!getDraft({ formDraftKey: FORM_DRAFT_KEY }))
+  const [showDiscardDraftModal, setShowDiscardDraftModal] = useState(false)
+
+  const [draft, setDraft] = useState<FormDraft | undefined>(
+    !readOnly ? getDraft({ formDraftKey: FORM_DRAFT_KEY }) : undefined)
+
+  const isSaveEnabled = !!draft && isEditorValid && !saving
+
+  const draftRef = useRef<FormDraft | undefined>()
+  draftRef.current = draft || undefined
+
+  useEffect(() => {
+    const saveToLocalStorage = () => {
+      if (draftRef.current) {
+        saveDraft({
+          formDraftKey: FORM_DRAFT_KEY,
+          draft: draftRef.current,
+          setSavingDraft
+        })
+      }
+    }
+
+    const draftSaveInterval = setInterval(saveToLocalStorage, 60000)
```

Comment:
best practice is to avoid using setInterval, and instead set up a timeout chain -- this prevents race conditions where the next action starts before the previous one finishes, and minimizes the impact if the cleanup fails.  Checkout lines 132-144 of SurveyView.tsx in ui-participant for an example

Reply:
Done in 8a38de640d3aea7ba1dc038d6fcf7aee2346a04a